src/frontend/screens/DeleteAudioBottomSheet.tsx

```
@@ -0,0 +1,87 @@
+import ErrorIcon from '../images/Error.svg';
+import {HeaderText} from '../sharedComponents/Text/HeaderText';
+import {BodyText} from '../sharedComponents/Text/BodyText';
+import {Button} from '../sharedComponents/Button';
+import {NativeRootNavigationProps} from '../sharedTypes/navigation';
+import {RED} from '../lib/styles';
+import {useDraftObservation} from '../hooks/useDraftObservation';
+
+const m = defineMessages({
+  title: {
+    id: 'screens.DeleteAudioBottomSheet.Title',
+    defaultMessage: 'Delete?',
+  },
+  description: {
+    id: 'screens.DeleteAudioBottomSheet.Description',
+    defaultMessage:
+      'Your Audio Recording will be permanently deleted. This cannot be undone.',
+  },
+  delete: {
+    id: 'screens.DeleteAudioBottomSheet.delete',
+    defaultMessage: 'Delete',
+  },
+  cancel: {
+    id: 'screens.DeleteAudioBottomSheet.cancel',
+    defaultMessage: 'Cancel',
+  },
+});
+
+export const DeleteAudioBottomSheet = ({
+  navigation,
+  route,
+}: NativeRootNavigationProps<'DeleteAudioBottomSheet'>) => {
+  const {formatMessage} = useIntl();
+  const {goBack, navigate} = navigation;
+  const {deleteAudio} = useDraftObservation();
+
+  return (
+    <BottomSheetWrapper>
+      <View style={styles.container}>
+        <View style={{alignItems: 'center'}}>
+          <ErrorIcon />
+          <HeaderText
+            variant="header2"
+            style={{marginTop: 20, textAlign: 'center'}}>
+            {formatMessage(m.title)}
+          </HeaderText>
+          <BodyText style={{marginTop: 10, textAlign: 'center'}}>
+            {formatMessage(m.description)}
+          </BodyText>
+        </View>
+        <View style={{width: '100%'}}>
+          <Button
+            fullWidth
+            color="dark"
+            style={{backgroundColor: RED, marginTop: 30}}
+            onPress={() => {
+              deleteAudio(
+                route.params.uri,
+                route.params.screenToPopToAfterDelete === 'ObservationEdit',
```

wondering if it makes sense to have this be derived either from the `uri` e.g. `uri.startsWith('https://')` or just another more explicit nav prop that's provided when navigating to this sheet e.g. `audioIsSaved: boolean`. the latter is probably better than the former in terms of clarity.

---

I actually wonder if we should just pass a  `onPressDeleteAudio:()=>void`? I know react navigation suggest passing only serializable objects in params, but that is for deep linking and state persistence. We are not persisting state and if a user deep links to a delete screen, I think we should throw an error (it would be bad ui to deep link to an action)

---

Looking into this, its only a suggestion by react navigation. So I think it does make sense to pass the onDeleteFunction here. It will make the bottom sheet agnostic, and not rely on the navigation context

---

yeah i thinking about suggesting something like that but then i remembered recommendation from react navigation. i wouldn't suggest going against that recommendation but as you say, it would technically work. i just worry about having to remember about a one-off case where we do this differently compared to other screens.

---

Agreed and perhaps we should just agree that the bottom sheets should just take a function. Currently ALL the bottom sheets are designed to take an action based on its parent. So if all the bottom sheets are taking a function in params it no longer becomes a one-off case.

This would align well with our principle of maintainability. The bottom sheets actions will be agnostic and allows for their actions to be swapped easier without having to rely on the navigation context

---

I did this here: https://github.com/digidem/comapeo-mobile/pull/958/commits/c6787289d487b5178d14297e3f18aeebf7afd260

Lets talk about it tomorrow at the app dev meeting